stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -198,101 +198,101 @@ namespace {
         : Type(type), Proto(proto) {
       assert(type);
     }
+
+    friend llvm::hash_code hash_value(const ConformanceCacheKey &key) {
+      return llvm::hash_combine(key.Type, key.Proto);
+    }
   };
 
   struct ConformanceCacheEntry {
   private:
     const Metadata *Type; 
     const ProtocolDescriptor *Proto;
-    std::atomic<const ProtocolConformanceDescriptor *> Description;
-    std::atomic<size_t> FailureGeneration;
+    std::atomic<const WitnessTable *> Witness;
```

Comment:
Does this still need to be atomic?  I think maybe it was atomic before only because entries were updated in-place occasionally.  Since you now flush the entire table anytime there's a change that might affect the cache contents, that means the cache entries are immutable and so their fields do not need to be atomic?

Reply:
In fact, I think it might be entirely const:  `const WitnessTable const * Witness`??

Reply:
I don't think making the field itself `const` buys you anything useful, since the entire `struct` will be `const` when you look it up using `snapshot.find()`. But making this non-atomic is a good idea!